### PR TITLE
Checks if it's an obelisk project

### PR DESCRIPTION
**Pull request by elgusy**
Checks if it is an obelisk project (created with ob init). If it is the case dispatches the appropriate repl command (ob repl) in an appropriate project root directory.

### CODE REVIEW

File: dante.el [hunk 1]
```diff
@@ -88,21 +88,47 @@ will be in different GHCi sessions."
...
+      (let* ((obelisk-root-folder (--take-while (-none? (lambda (a) (string= it a)) '("backend" "common" "frontend"))
+						file-dir-as-list))
+	     (project-root (string-join (-concat '("") obelisk-root-folder `("")) "/")))
+	(if (gethash project-root obelisk-projects-dirs nil)
+	    project-root
+	  (when (-all? (lambda (folder-name)
+			 (-> (-concat '("") obelisk-root-folder `(,folder-name ""))
+			     (string-join "/")
+			     (file-directory-p)))
+		       '("backend" "common" "config" "frontend" "static"))
```
Comment (srid) [inline dante.el]: A better check would be to see if the `.obelisk/impl` directory exists. All obelisk projects have it by design; that's what the (project-specific) `ob` commands use.

Comment (elgusy) [reply dante.el]: @srid, hi, yes much better. I didn't see that directory :) thanks. I've changed that line

File: dante.el [hunk 2]
```diff
@@ -88,21 +88,43 @@ will be in different GHCi sessions."
...
+				     (dante-buffer-file-name)))))))
+
+(defun obelisk-project-root ()
+  "Checks if a source file is a 'ob init' project's file stucture. In case it is, returns project's root
+   directory, also saves that directory in `obelisk-projects-dirs' set data structure so that `dante-repl-by-file'
+   can check faster is the project it is an obelisk project "
+  (let* ((file-dir-as-list (--filter (not (string= "" it)) (split-string (file-name-directory default-directory) "/")))
+	 (obelisk-folder-stucture? (--any? (-contains? file-dir-as-list it) '("backend" "common" "frontend"))))
+    (when obelisk-folder-stucture?
+      (let* ((obelisk-root-folder (--take-while (-none? (lambda (a) (string= it a)) '("backend" "common" "frontend"))
```
Comment (srid) [inline dante.el]: This should instead check for `.obelisk/impl` as well?

Comment (srid) [reply dante.el]: Hmm, I don't understand. What would happen if you check for existence of just `.obelisk/impl` (instead of those source folders)?

Comment (elgusy) [reply dante.el]: To check the existence of the folder `.obelisk/impl` I need to provide its parent directory, the obelisk root folder, in my above example, - `learning-reflex`. To track that root folder I check if (common, backend, frontend) exists in the opened source file's path.

**Comment by bidigo** [comment]
Any timeline when/if is this going to be merged?

**Comment by jyp** [comment]
@bidigo This PR was a bit to ad hoc. I wanted to have a generic way to search for root projects, which I've now implemented. You should be able to do what you want with a customization of `dante-methods-alist`. Let me know.

**Pull request closed without merging**